metsim/metsim.py

```diff
@@ -18,37 +18,61 @@ class MetSim(object):
     work can be done while IO is happening.
     """
 
-    def __init__(self, domain:str, params:dict):
+    # Class variables
+    methods = {'mtclim': mtclim}
+    params = {
+        "method":'mtclim',
+        "domain":'',
+        "out_dir":'./results/',
+        "start":'',
+        "stop":'',
+        "time_step":'',
+        "out_format":'netcdf',
+        "site_elev": 0.,
+        "base_elev": 0.,
+        "t_max_lr": 0.0065,
+        "t_min_lr": 0.0065,
+        "site_isoh": None,
+        "base_isoh": None,
+        "site_lat": 0.,
+        "site_slope": 0.,
+        "site_aspect": 0.,
+        "site_east_horiz": 0.,
+        "site_west_horiz": 0.
+    }   
+
+
+    def __init__(self, params:dict):
         """
         Constructor
         """
-        # Record parameters and broadcast to the config module
-        self.params = params
-        configuration.update(params) 
+        # Record parameters 
+        self.update(params)
 
         # Get the necessary information from the domain 
-        domain = Dataset(configuration.PARAMS['domain'], 'r')
+        domain = Dataset(MetSim.params.get('domain', None), 'r')
```

Comment:
won't `Dataset` raise an error if it gets `None`? I would just let python raise a `KeyError` with `Dataset(MetSim.params['domain'], 'r')` so as not to confuse the error.